Zoo/MainZoo.cs

```diff
@@ -0,0 +1,287 @@
+
+
+        public event notify zooEvents;
+
+        public MainZoo(List<Animal> animals, List<Worker> workers)
+        {
+            if (CheckZoo(animals))
+            {
+                this.animals = animals;
+                tourId = 1;
+                this.workers = workers;
+                workersDictStart = new Dictionary<int, List<Worker>>();
+                workersDictFinish = new Dictionary<int, List<Worker>>();
+                tours = new List<Tour>();
+                AddWorkersToDicts();
+                zooEvents += new notify(AddAndRemoveTours);
+                zooEvents += new notify(PrintStartAndFinish);
+                zooIncome = 0;
+                Manager.LogEvents("Zoo created.");
+            }
+            else
+                Manager.LogEvents("Can't create Zoo.");
+        }
+
+        public bool CheckZoo(List<Animal> animalArr)
+        {
+            int landCounter = 0;
+            int seaCounter = 0;
+            int skyCounter = 0;
+            int mixCounter = 0;
+            List<Animal> tempAnimals = new List<Animal>();
+
+            foreach (Animal animal in animalArr)
+            {
+                for(int i = 0; i < tempAnimals.Count; i++)
+                {
+                    if(animal.GetType().ToString() == animalArr[i].GetType().ToString())
+                    {
+                        Manager.LogEvents("Found two the same.");
+                        return false;
+                    }
+                }
+                switch (animal)
+                {
+                    case LandAnimals:
+                        landCounter++;
+                        break;
+                    case SeaAnimals:
+                        seaCounter++;
+                        break;
+                    case SkyAnimals:
+                        skyCounter++;
+                        break;
+                    case MixedAnimal:
+                        mixCounter++;
+                        break;
+                }
+                tempAnimals.Add(animal);
+            }
+            return landCounter >= 3 && seaCounter >= 3 && skyCounter >= 3 && mixCounter >= 3;
+        }
+
+        public void AddWorkersToDicts()
+        {
+            foreach(Worker w in workers)
+            {
+                foreach(int time in w.WorkingTimes)
+                {
+                    AddToDictsHelper(ref workersDictStart, w, time);
+                    AddToDictsHelper(ref workersDictFinish, w, time + w.WorkingInterval);
+                }
+            }
+        }
+
+        public void AddToDictsHelper(ref Dictionary<int, List<Worker>> dict, Worker w, int time)
+        {
+            try
+            {
+                dict[time].Add(w);
+            }
+            catch (Exception e)
+            {
+                List<Worker> temp = new List<Worker>();
+                temp.Add(w);
+                dict.Add(time, temp);
+            }
+        }
+
+        public void PrintStartAndFinish(int time)
```

Comment:
I sense a copy+c, copy+v here. meaning you have code coupling. Same functionality can be done once with a function